create separate 'development only' requirements file

bhamail opened the pull request:
Collect "development only" requirements into a new file, and use that for various CI tasks.

cc @bhamail / @DarthHater

DarthHater commented on requirements-dev.txt:
```diff
@@ -0,0 +1,3 @@
+pylint
```
Ideally we'd have versions for all these.

DarthHater commented:
@bhamail I think to generate the requirements.txt it would be good to do the following:

- start a empty venv
- pip install the packages we need
- pip freeze > requirements-dev.txt

This SHOULD get versions to lock to, and also gather any transitive dependencies!

DarthHater commented:
ALSO, I did some quick reading last night, and people are suggesting instead of `pip` to look at using `pipenv` nowadays, as it has some amount of differentiation for dev dependencies

DarthHater reviewed (changes requested):
Suggested change to how we generate the requirements-dev.txt

bhamail commented:
Added initial `pipenv` changes, but ran into issues getting it to work with CI build. So, I updated the `requirements-dev.txt` file with the versions resolved via `pipenv`. I also added the `pipenv` config files: `Pipfile` and `Pipfile.lock`. I noticed the version of dev dependencies (installed via `pipenv install pylint --dev`) are only locked to versions in the .lock file, but are wildcards in the `Pipfile`.

I think these changes get us closer to using `pipenv`, while also updating to use a shared `dev dependencies` list. We can revisit full conversion to `pipenv` when time permits. (PR's welcome from any `pipenv` experts! :)

DarthHater reviewed (approved):
I'm a lil bit iffy on the Pipenv files going in but I don't see a full issue with doing so.
